src/Stratis.FederatedPeg.Tests/LeaderProviderTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NBitcoin;
+using NSubstitute;
+using Stratis.FederatedPeg.Features.FederationGateway;
+using Xunit;
+
+namespace Stratis.FederatedPeg.Tests
+{
+    public class LeaderProviderTests
+    {
+        private readonly IFederationGatewaySettings federationGatewaySettings;
+
+        private readonly LeaderProvider leaderProvider;
+
+        private List<string> leaderPubkeys;
+
+        public LeaderProviderTests()
+        {
+            this.federationGatewaySettings = Substitute.For<IFederationGatewaySettings>();
+            this.federationGatewaySettings.FederationPublicKeys.Returns(this.BuildLeaderPubKeys());
+            this.leaderProvider = new LeaderProvider(this.federationGatewaySettings);
+        }
+
+        [Fact]
+        public void UpdatingBlockHeightCarriesOutLeaderPubKeysRoundRobin()
+        {
+            var output = new List<PubKey>();
```

Comment:
I would do the checks in the same loop as the output then you don't need this output to store things, up to you.

Reply:
The extra output is to check there was one round of changes to the leader.  There not much cost storing this and it's a test.  Also it's easier to follow then writing it out in the first loop, when having to cater for index values > 5 etc.